tests/test.cpp

```diff
@@ -832,3 +830,29 @@ TEST_CASE("Sort on disk")
         delete[] memory;
     }
 }
+
+TEST_CASE("bitfield_index-simple")
+{
+    std::vector<bool> const bitfield{true, true, false, true};
+    bitfield_index const idx(bitfield);
+    CHECK(idx.lookup(0, 0) == std::pair<uint64_t, uint64_t>{0,0});
+    CHECK(idx.lookup(0, 1) == std::pair<uint64_t, uint64_t>{0,1});
+
+    CHECK(idx.lookup(0, 3) == std::pair<uint64_t, uint64_t>{0,2});
+
+    CHECK(idx.lookup(1, 0) == std::pair<uint64_t, uint64_t>{1,0});
+    CHECK(idx.lookup(1, 2) == std::pair<uint64_t, uint64_t>{1,1});
+    CHECK(idx.lookup(3, 0) == std::pair<uint64_t, uint64_t>{2,0});
+}
+
+TEST_CASE("bitfield_index-use index")
+{
+    std::vector<bool> bitfield(1024 * 1024, false);
+    CHECK(bitfield.size() == 1048576);
+    bitfield[1048576 - 3] = true;
+    bitfield[1048576 - 2] = true;
+    bitfield[1048576 - 1] = true;
+    bitfield_index const idx(bitfield);
+    CHECK(idx.lookup(1048576 - 3, 1) == std::pair<uint64_t, uint64_t>{0,1});
+    CHECK(idx.lookup(1048576 - 2, 1) == std::pair<uint64_t, uint64_t>{1,1});
+}
```

Comment:
Suggest boundary condition tests for `bitfield_index`:

// One past bucket
idx.lookup(pos=0, offset=kIndexBucket)

// One past bitfield.size()

Reply:
I used to have tests for unset bits, that they would throw. But this is a pretty hot function so I backed out that check. Now it's up to the caller to only use valid positions and offsets (hence the asserts)